htdocs/admin/mails.php

```diff
@@ -737,14 +737,14 @@ function initfields()
 
 		print '<tr class="oddeven"><td>'.$langs->trans('MAIN_MAIL_DEFAULT_FROMTYPE').'</td>';
 		print '<td>';
-		if ($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE === 'robot') {
+		if (!empty($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE) === 'robot') {
```

Comment:
!empty($conf->global->MAIN_MAIL_DEFAULT_FROMTYPE) is a boolean. Comparing with 'robot' will always be false.